### PR TITLE
Add page listing events in the deputy district

### PR DESCRIPTION
**Pull request by illusionOfParadise**
https://app.clubhouse.io/enmarche/story/4435/etq-d%C3%A9put%C3%A9-dans-la-section-evenements-je-ne-vois-que-les-events-de-ma-zone-g%C3%A9r%C3%A9e-circonscription

### CODE REVIEW

File: tests/Referent/ManagedEventsExporterTest.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function testExportAsJsonWithNoOrganizer()
         $expectedDate = $event->getBeginAt()->format('Y-m-d');
 
         $this->assertSame(
-            '[{"id":14,"name":{"label":"Meeting #11 de Brooklyn","url":"'.$this->hosts['scheme'].':\/\/'.$this->hosts['app'].'\/evenements\/'.$expectedDate.'-meeting-11-de-brooklyn"},"beginAt":"'.$expectedBeginAt.'","category":"Marche","postalCode":"10019","organizer":"un ancien adh\u00e9rent","participants":0}]',
+              '[{"id":14,"name":{"label":"Meeting #11 de Brooklyn","url":"'.$this->hosts['scheme'].':\/\/'.$this->hosts['app'].'\/evenements\/'.$expectedDate.'-meeting-11-de-brooklyn"},"beginAt":"'.$expectedBeginAt.'","category":"Marche","postalCode":"10019","organizer":"un ancien adh\u00e9rent","participants":0,"type":"Comit\u00e9"}]',
```
Comment (Remg) [inline tests/Referent/ManagedEventsExporterTest.php]: extra indentation

File: src/Repository/EventRepository.php [hunk 2]
```diff
@@ -208,6 +209,25 @@ public function findManagedBy(Adherent $referent): array
...
     }
 
+    /**
+     * @return BaseEvent[]
+     */
+    public function findAllInDistrict(District $district): array
+    {
+        return $this->_em->getRepository(BaseEvent::class)->createQueryBuilder('e')
+            ->select('e', 'o')
+            ->leftJoin('e.organizer', 'o')
```
Comment (Remg) [inline src/Repository/EventRepository.php]: is this leftJoin needed ?

File: app/config/services/services.xml [hunk 3]
```diff
@@ -141,6 +141,7 @@
         <service id="AppBundle\Command\ReferentTagInitializeCommitteeEvent"/>
         <service id="AppBundle\Referent\ReferentTagManager" />
         <service id="AppBundle\Referent\ManagedCommitteesExporter" public="true" />
+        <service id="AppBundle\Referent\ManagedEventsExporter" public="true" />
```
Comment (ottaviano) [inline app/config/services/services.xml]: normally, you don't need to use public here, 
you can redeclare the same service with prefix `test.AppBundle\Referent\ManagedEventsExporter` in `config_test.yml` with `public: true` config.

File: src/Repository/EventRepository.php (hunk 2 above)
Comment (illusionOfParadise) [reply src/Repository/EventRepository.php]: Yes, the organizer data is used in the events exporter.

Comment (Remg) [reply src/Repository/EventRepository.php]: :zipper_mouth_face: :100:

File: src/Repository/EventRepository.php [hunk 4]
```diff
@@ -208,6 +209,25 @@ public function findManagedBy(Adherent $referent): array
...
+            ->where('e.published = :published')
+            ->andWhere("ST_Within(ST_GeomFromText(CONCAT('POINT(',e.postAddress.longitude,' ',e.postAddress.latitude,')')), :district) = 1")
```
[the remark this answers is not in the document]
Comment (illusionOfParadise) [reply src/Repository/EventRepository.php]: 1. Yes and it does not work.
2. I will try, thanks.
3. Yes.

[the remark this answers is not in the document]
Comment (ottaviano) [reply src/Repository/EventRepository.php]: for 1., it works fine for me:
```sql
SELECT a.address_city_name, a.address_postal_code, a.address_country, d.countries, d.name FROM adherents AS a 
JOIN districts AS d
WHERE
WITHIN(POINT(address_longitude, address_latitude), d.geo_shape);
```
what is your error?

**Pull request merged**